packages/eventual-send/src/index.d.ts

```diff
@@ -21,45 +25,63 @@ type ReturnType<T> = T extends (...args: any[]) => infer T ? T : any;
 
 interface EHandler<T> {
   get?: (p: T, name: Property) => any;
-  applyMethod?: (p: T, name?: Property, args: unknown[]) => any;
+  getSendOnly?: (p: T, name: Property) => void;
+  applyFunction?: (p: T, args: unknown[]) => any;
+  applyFunctionSendOnly?: (p: T, args: unknown[]) => void;
+  applyMethod?: (p: T, name: Property | undefined, args: unknown[]) => any;
+  applyMethodSendOnly?: (
+    p: T,
+    name: Property | undefined,
+    args: unknown[],
+  ) => void;
 }
 
 type HandledExecutor<R> = (
-  resolveHandled: (value?: R) => void,
+  resolveHandled: (() => void) & ((value: R) => void),
```

Comment:
Interesting. I assume this solves a problem of matching arity.

Reply:
Why not

```suggestion
  resolveHandled: (value?: R) => void,
```
?

After this change, `yarn lint` and vscode are both still silently happy.